server/src/main/java/com/microsoft/java/bs/core/internal/gradle/GradleApiConnector.java

```diff
@@ -153,18 +170,20 @@ public StatusCode runTasks(URI projectUri, ProgressReporter reporter, String...
   /**
    * request Gradle to run tests.
    */
-  public StatusCode runTests(URI projectUri,
+  public StatusCode runTests(
+      URI projectUri,
       Map<BuildTargetIdentifier, Map<String, Set<String>>> testClassesMethodsMap,
       List<String> jvmOptions,
       List<String> args,
       Map<String, String> envVars,
       BuildClient client, String originId,
-      CompileProgressReporter compileProgressReporter) {
+      CompileProgressReporter compileProgressReporter
+  ) {
 
     StatusCode statusCode = StatusCode.OK;
     ProgressReporter reporter = new DefaultProgressReporter(client);
     try (ProjectConnection connection = getGradleConnector(projectUri).connect()) {
-      String gradleVersion = getGradleVersion(connection);
+      String gradleVersion = getGradleVersion(projectUri);
```

Comment:
Could you reuse the connection here? For example, overloda the `getGradleVersion()`?

Reply:
Addressed in commit [fbf550b](https://github.com/microsoft/build-server-for-gradle/pull/165/commits/fbf550b288559f5f92a474307885ce56e417d962)